runners/core-construction-java/src/main/java/org/apache/beam/runners/core/construction/SdkComponents.java

```diff
@@ -261,14 +263,20 @@ public String registerCoder(Coder<?> coder) throws IOException {
    * return the same unique ID.
    */
   public String registerEnvironment(Environment env) {
+    String environmentId;
     String existing = environmentIds.get(env);
     if (existing != null) {
-      return existing;
+      environmentId = existing;
+    } else {
+      String name = uniqify(env.getUrn(), environmentIds.values());
+      environmentIds.put(env, name);
+      componentsBuilder.putEnvironments(name, env);
+      environmentId = name;
     }
-    String name = uniqify(env.getUrn(), environmentIds.values());
-    environmentIds.put(env, name);
-    componentsBuilder.putEnvironments(name, env);
-    return name;
+    if (defaultEnvironmentId == null) {
```

Comment:
+1

Reply:
Created a separate ticket: https://issues.apache.org/jira/browse/BEAM-9425

Reply:
Why should this be a separate ticket ? Can't we just change method to registerEnvironment(Environment env, boolean default)
and change above to

sdkComponents.registerEnvironment(	        
    Environments.createOrGetDefaultEnvironment(portablePipelineOptions), true);

Reply:
If we change the signature of `registerEnvironment`, a number of test files (*TranslationTest, *RunnerTest) also need to be touched. I think it will create unnecessary noise in this PR.

Reply:
Ok. Let's still do this immediately after this one though so that we do not forget about it.

Reply:
This is a case where doing it in a different commit in the same PR could make sense. But I'm OK with deferring as long as it happens soon.